core/divesite.c

```diff
@@ -382,6 +382,29 @@ void purge_empty_dive_sites(struct dive_site_table *table)
 	}
 }
 
+void add_dive_to_dive_site(struct dive *d, struct dive_site *ds)
+{
+	int idx;
+	if (d->dive_site == ds)
+		return;
+	if (d->dive_site)
+		fprintf(stderr, "Warning: adding dive that has dive site set to dive site\n");
```

Comment:
that is an, err, interesting message...

Reply:
Reworder - better?
[This is a shouldn't happen warning - so the fact that it is printed is probably more of a concern than it being cryptic.]